tiledb/libtiledb.pyx

```diff
@@ -4227,14 +4227,22 @@ cdef class Query(object):
     """
 
     def __init__(self, array, attrs=None, coords=False,
-                 order='C', use_arrow=None, return_arrow=False):
+                 order=None, use_arrow=None, return_arrow=False):
         if array.mode != 'r':
             raise ValueError("array mode must be read-only")
         self.array = array
         self.attrs = attrs
         self.coords = coords
-        self.order = order
         self.return_arrow = return_arrow
+
+        if order == None:
+            if array.schema.sparse:
+                self.order = 'U'
+            else:
+                self.order = 'C'
```

Comment:
Is this suppose to be column or row major ordering?

Reply:
Same as above